### PR TITLE
[tests-only] [full-ci] Added `Then` steps to improve checks in tests for apiFavorites suite

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of `When` steps.

**Suites covered:**
- `apiFavorites`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiFavorites/favorites.feature [hunk 1]
```diff
@@ -78,13 +81,16 @@ Feature: favorite
   Scenario Outline: Favorite and unfavorite a file
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiFavorites/favorites.feature:81]: I think the favoriting scenario is already there in line 60, so this scenario should be like
```suggestion
  Scenario Outline: Favorite an unfavorited a file
```
and afterward,
-  favoriting a file should be in `given` step.
-  unfavoriting a file should be in `when` step (as it is now)

Comment (grgprarup) [reply tests/acceptance/features/apiFavorites/favorites.feature]: Thanks a lot :+1:

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by SwikritiT — COMMENTED** [review]
lgtm

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by amrita-shrestha — COMMENTED** [review]
lgtm

**Review by SagarGi — COMMENTED** [review]
LGTM

**Review by kiranparajuli589 — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**